pkg/apis/servicecatalog/v1beta1/types.go

```diff
@@ -92,6 +92,61 @@ type ClusterServiceBrokerSpec struct {
 	// can be manually incremented by a user to manually trigger a relist.
 	// +optional
 	RelistRequests int64 `json:"relistRequests"`
+
+	// CatalogRestrictions allows adding restrictions onto Class/Plans on relist.
+	// +optional
+	CatalogRestrictions *ServiceClassCatalogRestrictions `json:"catalogRestrictions,omitempty"`
+}
+
+// *Requirements type strings have a special format similar to Label Selectors,
+// except the catalog supports only a very specific property set per type as
+// documented below.
+// The predicate format is expected to be `<property><conditional><requirement>`
+// Check the *Requirements type definition for which <property> strings will be allowed.
+// <conditional> is allowed to be one of the following: ==, !=, in, notin
+// <requirement> will be a string value if `==` or `!=` are used.
+// <requirement> will be a set of string values if `in` or `notin` are used.
+// Multiple predicates are allowed to be chained with a comma (,)
+
+// ClusterServiceClassRequirements represents a selector for classes used to filter.
+// Allowed property names:
+// name - the value set to ClusterServiceClass.Name
+// externalName - the value set to ClusterServiceClass.Spec.ExternalName
+type ClusterServiceClassRequirements string
+
+// ClusterServicePlanRequirements represents a selector for plans used to filter.
+// Allowed property names:
+// name - the value set to ClusterServiceClass.Name
+// externalName - the value set to ClusterServiceClass.Spec.ExternalName
+type ClusterServicePlanRequirements string
+
+// ServiceClassCatalogRestrictions contains the restrictions used to on catalog re-list.
+// Some examples of this object are as follows:
+//
+// This is an example of a whitelist on service externalName.
+// Goal: Only list Services with the externalName of FooService and BarService,
+// Solution: restrictions := ServiceClassCatalogRestrictions{
+// 		ServiceClass: "externalName in (FooService, BarService)",
+// }
+//
+// This is an example of a blacklist on service externalName.
+// Goal: Allow all services except the ones with the externalName of FooService and BarService,
+// Solution: restrictions := ServiceClassCatalogRestrictions{
+// 		ServiceClass: "externalName notin (FooService, BarService)",
+// }
+//
+// This whitelists plans called "Demo", and blacklists (but only a single element in
+// the list) a service and a plan.
+// Goal: Allow all plans with the externalName demo, but not AABBCC, and not a specific service by name,
+// Solution: restrictions := ServiceClassCatalogRestrictions{
+// 		ServiceClass: "name!=AABBB-CCDD-EEGG-HIJK",
+// 		ServicePlan: "externalName in (Demo),name!=AABBCC",
+// }
+type ServiceClassCatalogRestrictions struct {
+	// ServiceClass represents a selector for plans, used to filter catalog re-lists.
+	ServiceClass ClusterServiceClassRequirements `json:"serviceClass,omitempty"`
```

Comment:
I could do an array, makes it less easy to fat finger a long string.

As for the structured definition, k8s does both. The object that this turns into is non-trivial because it is pretty powerful and then we either leak the fact we are using label selectors or we have to duplicate the label selector code. 

I vote array of strings for now, if that pleases you

Reply:
Array of strings is fine for me.